data/types.go

```diff
@@ -226,6 +229,7 @@ type Targets struct {
 	Targets     TargetFiles      `json:"targets"`
 	Delegations *Delegations     `json:"delegations,omitempty"`
 	Custom      *json.RawMessage `json:"custom,omitempty"`
+	NewTarget   bool             `json:"-"`
```

Comment:
Maybe `NewTargets` is a more consistent name.

Reply:
It seems like this is just meant to check if the root is at the initial. 

Maybe it's best to stage the root and targets file in the same place that `NewTargets` is called? @ethan-lowman-dd suggested this here: https://github.com/theupdateframework/go-tuf/issues/272#issuecomment-1271733490

Consider replacing all the `data.NewRoot()` function calls in `repo.go` with a new function
```
func (r *Repo) NewRoot()  error {
	root := data.NewRoot()
        // This will stage the metadata, which means that there won't be any further increments.
        return r.setMeta("root.json", root)
}
```

Reply:
You know, that's not a bad idea. Let me talk to @abs007 about this.

Reply:
> > Could you confirm if this is what you mean:
> > 1.1: Have tests to check if NewRoot/NewTarget returns a struct with`Version`: 1 1.2: To check if the new`NewRoot` bool field doesn't cause clashes for when files are loaded in?
> 
> Let's sync over Zoom this week?

Let's do that 👍

Reply:
We tried this approach just now, and unfortunately, due to a `repo.setMeta()->repo.root()` recursion, we are unable to use this trick, at least in the `repo.root()` function. I don't see a better way around this than to rework the API. WDYT?

Reply:
@asraa could we get your feedback? Thanks.

Reply:
Got it. The problem with the recursion is because `setMeta()` is trying to retrieve the `root()` to gather the signers. If there's no root yet, then you shouldn't have anything to sign. I actually think it SHOULDN'T fail in that case (if there's no signers). 

If you fix the recursion, the tests still fail because the tests try to do repo operations like create keys when `repo.Init` was never called.

Should `Init` always be called before trying to add root keys and targets? I think it should...

Reply:
I think so. My concern is that if the API cannot handle such a simple change (e.g., going from v0 to v1), then I'm worried about refactoring it just to fix the issue in hand now, only to find unforeseen consequences later 🙁 Personally, I would much rather redesign and rewrite the API instead of trying to fix it now, but this might be the wrong thing to do if no one has the time to work on the better API (not to mention that the kludge in this PR is hacky). What do others think? @znewman01 @mnm678 

Also, more importantly, how important is this issue to fix right now? 🙂

Reply:
Yeah: agree, it's not so important to handle right now because there are ways the user won't encounter the issue.

I managed to get a patch & tests working that removed the recursion so long as a root is staged whenever a repo is initialized... I can share it with you, but yes: it involves that `NewRepo` or `Init` must stage a new root file (I think that's a safe assumption, but right now clients are probably unsure when Init is requried. My understanding is that it's required for a new repo, where `NewRepo` can be created off an existing store).

Reply:
That's great! Would you like to handover the patch here so that @abs007 and I could help push it thru?

Reply:
https://gist.github.com/asraa/47a820c3c8e593416d2d67680ea48107

here it is! you have to regen the client interop code though too, but I tested it to work

Right now the least changes were done staging a new root when you call `NewRepo`, but ideally I think that belongs in `Init`. But tests usually don't call `Init`, aha

Reply:
Thanks very much! @abs007 Do you think you have all the pieces? Happy to meet this weekend on the same day/time to help.

Reply:
So, I tried exploring this call stack: ` setMeta() -> signersforRole() -> getSignersInDB() -> GetSigners()` and noticed this [line](https://github.com/theupdateframework/go-tuf/blob/master/local_store.go#L150) in `local_store.go`. Aren't we supposed to be returning `nil, err` because we do expect an err in [here](https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L909) it seems.

Reply:
Also, I tried exploring this call stack: ` setMeta() -> signersforRole() -> getSignersInDB() -> GetSigners()` and noticed this [line](https://github.com/theupdateframework/go-tuf/blob/master/local_store.go#L150) in `local_store.go`. Aren't we supposed to be returning `nil, err` because we do expect an err in [here](https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L909) it seems.

It seems that this might be allowing the empty db obtained from [here](https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L747)

Reply:
@asraa  wdyt?

Reply:
> Aren't we returning an empty db here? https://gist.github.com/asraa/47a820c3c8e593416d2d67680ea48107#file-gistfile1-txt-L67

Yes, but we're only returning the empty DB when the root is Unitialized (as in, just this first instance)

Reply:
> Also, I tried exploring this call stack: setMeta() -> signersforRole() -> getSignersInDB() -> GetSigners() and noticed this [line](https://github.com/theupdateframework/go-tuf/blob/master/local_store.go#L150) in local_store.go. Aren't we supposed to be returning nil, err because we do expect an err in [here](https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L909) it seems.

I think this is returning the empty list of signers though, which is OK because we don't require any signers. See how these return nil as well https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L923 -- as in, it's not an error that we don't have any signers available.

Reply:
Yeah, makes sense. That being said, why does [this](https://github.com/theupdateframework/go-tuf/blob/master/repo.go#L909) expect an error though if we'll never return one?